storage/tests/ledger_validation.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use snarkos_storage::{storage::rocksdb::RocksDB, LedgerState};
+use snarkvm::{dpc::testnet2::Testnet2, prelude::Block};
+
+use rand::{thread_rng, Rng, SeedableRng};
+use rand_xorshift::XorShiftRng;
+
+use std::{fs, path::Path};
+
+const NUM_BLOCKS: usize = 1_000;
+const NUM_CHECKS: usize = 100;
+
+#[test]
+#[ignore = "takes a while to run (most of which is deserialization)"]
+fn test_ledger_validation() {
+    let mut test_blocks = fs::read(format!("benches/blocks_{}", NUM_BLOCKS)).expect(&format!("Missing the test blocks file"));
+    let blocks: Vec<Block<Testnet2>> = bincode::deserialize(&mut test_blocks).expect("Failed to deserialize a block dump");
+    assert_eq!(blocks.len(), NUM_BLOCKS - 1);
```

Comment:
Maybe just comment on the `len` not including genesis here?

Reply:
Fair enough, as this test is also missing the comment on how the test blocks were created. I'll include that soon :tm:.